executor/insert_common.go

```diff
@@ -249,21 +249,34 @@ func (e *InsertValues) handleErr(col *table.Column, val *types.Datum, rowIdx int
 		return nil
 	}
 
-	if types.ErrDataTooLong.Equal(err) {
-		return resetErrDataTooLong(col.Name.O, rowIdx+1, err)
+	// Convert the error with full messages.
+	var (
+		colTp   byte
+		colName string
+	)
+	if col != nil {
```

Comment:
if `col == nil` does  following code will generate error message use unspecified type and '' column name?

Reply:
yes...